contributors/design-proposals/admission_control_extension.md

```diff
@@ -0,0 +1,594 @@
+                  WATCH /pods/1                         v                    |
+                      |                              change resources        |
+                      |                              clear initializer       |
+                      |            validate <------- PUT /pods/1             |
+                      |            admission(init):                          |
+                      |              check authz                             v
+                      |            save etcd ---------------------------> observe
+                      |                                                   change env vars
+                      |                                                   clear initializer
+                      |            validate <---------------------------- PUT /pods
+                      |            admission(init):
+                      v            check authz
+                  observe <------- save etcd
+response <------- handle
+```
+
+An example flow where initialization fails:
+
+```
+Client            APIServer (req)  APIServer (other) Initializer 1        Initializer 2
+----------------- ---------------- ----------------- -------------------- -------------------
+
+                                   listen <--------- WATCH /pods?init=0
+                                   listen <-------------|---------------- WATCH /pods?init=0
+                                                        |                    |
+POST /pods -----> validate                              |                    |
+                  admission(init):                      |                    |
+                    default                             v                    |
+                  save etcd -----------------------> observe                 |
+                  WATCH /pods/1                         v                    |
+                      |                              change resources        |
+                      |                              clear initializer       |
+                      |            validate <------- PUT /pods/1             |
+                      |            admission(init):                          |
+                      |              check authz                             v
+                      |            save etcd ---------------------------> observe
+                      |                                                   failed to retrieve object
+                      |                                                   set result to failure
+                      |            validate <---------------------------- PUT /pods
+                      |            admission(init):
+                      |              check authz
+                      v            signal failure
+                  observe <------- save etcd
+response <------- handle               |
+                                       v
+                                   delete object
```

Comment:
Good question.  I think initializers should be able to set finalizers, and initializers should be allowed to create other actions as the result of a create (even if it might be a bad design).

We might also want to allow finalizers to be set directly via the dynamic admission config vs having to write an initializer for it.